File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
  	//品牌服务层
    	app.service('brandService',function($http){
    		//读取列表数据绑定到表单中
    		this.findAll=function(){
    			return $http.get('/brand/findAll.do');
    		}
    		//分页 
    		this.findPage=function(page,size){
    			return $http.get('/brand/findPage.do?page='+page+'&size='+size);
    		}
    		//添加
    		this.add=function(entity){
    			return $http.post('/brand/add.do',entity);
    		}
    		//更新数据
    		this.update=function(entity){
    			return  $http.post('/brand/update.do',entity);
    		}
    		
    		//查询实体
    		this.findOne=function(id){
    			return $http.get('/brand/findOne.do?id='+id);
    		}
    		
    		//批量删除
    		this.dele=function(id){
    			return $http.get("/brand/delete.do?ids="+id);
    		}
    		
    		//模糊查询
    		this.search=function(page,size,searchEntity){
    			return $http.post('/brand/search.do?page='+page+'&size='+size,searchEntity);
    		}
    		
    		
    		//下拉列表数据
        	this.selectOptionList=function(){
        		return $http.get('../brand/selectOptionList.do');
        	}
    		
    	});
    	
    	
    	

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

/**
 * 品牌服务层接口
 * 
 * @author liucx
 *
 */
public interface BrandService {

	/**
	 * 返回全部列表
	 * 
	 * @return
	 */
	public List<TbBrand> findAll();

	/**
	 * 返回分页列表
	 * 
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(Integer pageNum, Integer pageSize);

	/**
	 * 添加
	 * 
	 * @param brand
	 */
	public void add(TbBrand brand);

	/**
	 * 修改
	 * 
	 * @param brand
	 */
	public void update(TbBrand brand);

	/**
	 * 根据 ID 获取实体
	 * 
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);

	/**
	 * 批量删除
	 * 
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 模糊查询
	 * 
	 * @param brand
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult search(TbBrand brand, Integer pageNum, Integer pageSize);

	/**
	 * 品牌下拉框数据
	 * 
	 * @return
	 */
	List<Map> selectOptionList();
}
